### PR TITLE
comments resolved

### PR DESCRIPTION
**Pull request by sehrnadeem1**
Session comments resolved

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -33,10 +42,25 @@ def update
 
   #DELETE /items/:id
   def destroy
-    if @item.destroy
-      flash[:notice] = 'Item successfully deleted'
+    item_found, item_deleted = false, false
+    orders = Order.all
```
Comment (HassanAkbar) [inline app/controllers/items_controller.rb]: use the query

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -33,10 +42,25 @@ def update
...
+    orders.each do |order|
+      orderItems = order.order_items
+      orderItems.each do |orderItem|
+        item_found = true if orderItem.item == @item
+      end
+    end
+
+    if item_found
+      item_deleted = true if @item.update(active: 0)
```
Comment (HassanAkbar) [inline app/controllers/items_controller.rb]: (active: false)

File: app/controllers/order_items_controller.rb [hunk 3]
```diff
@@ -2,65 +2,68 @@ class OrderItemsController < ApplicationController
   load_and_authorize_resource :order
   load_and_authorize_resource :order_item, through: :order
 
+  #GET /order/:order_id/order_items
+  def index
+    @order_items = @order_items.order(created_at: :desc).paginate(page: params[:page], per_page: 4)
```
Comment (HassanAkbar) [inline app/controllers/order_items_controller.rb]: make a file app_constants in config_initializers and add a constant :
LISTING_PAGINATION_SIZE = 4
Use the pagination size everywhere

File: app/controllers/order_items_controller.rb [hunk 4]
```diff
@@ -2,65 +2,68 @@ class OrderItemsController < ApplicationController
...
+    respond_to do |format|
+      format.html
+    end
+  end
+
   #GET /order/:order_id/order_items/new
   def new
-    @items = Item.all
+    @items = Item.where(active: true)
```
Comment (HassanAkbar) [inline app/controllers/order_items_controller.rb]: create a scope in item model

````ruby
scope :active, -> { where(active: true) }
`
Use here:
Item.active

File: app/controllers/order_items_controller.rb [hunk 5]
```diff
@@ -69,10 +72,17 @@ def update
 
   #DELETE /order/:order_id/order_items/:id
   def destroy
-    if @order_item.destroy
-      flash[:notice] = 'Order item successfully deleted.'
+    if @order_item.order.status != 'Completed'
```
Comment (HassanAkbar) [inline app/controllers/order_items_controller.rb:79]: constant maybe

File: app/controllers/order_items_controller.rb [hunk 6]
```diff
@@ -85,11 +95,26 @@ def order_item_params
   end
 
   def check_updated_quantity
```
Comment (HassanAkbar) [inline app/controllers/order_items_controller.rb]: model

File: app/controllers/waiters_controller.rb [hunk 7]
```diff
@@ -3,11 +3,43 @@ class WaitersController < ApplicationController
...
     end
   end
 
+  def edit
+    @waiter = User.find_by(id: params[:id])
+  end
+
+  def update
+    @waiter = User.find_by(id: params[:id])
+    if @waiter.update(waiter_params)
```
Comment (HassanAkbar) [inline app/controllers/waiters_controller.rb]: @waiter.present? &&

File: app/controllers/waiters_controller.rb [hunk 8]
```diff
@@ -3,11 +3,43 @@ class WaitersController < ApplicationController
...
+      redirect_to waiters_path
+    else
+      flash[:alert] = I18n.t(:waiter_update_fail, error: @waiter.errors.full_messages.to_sentence)
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @waiter = User.find_by(id: params[:id])
+    if @waiter.destroy
```
Comment (HassanAkbar) [inline app/controllers/waiters_controller.rb]: @waiter.present? &&

File: app/views/orders/_searchform.html.erb [hunk 9]
```diff
@@ -0,0 +1,33 @@
+<div class="modal-dialog" role="document">
+  <div class="modal-content">
+    <div class="modal-header">
+      <h4 class="modal-title text-center" id="myModalLabel">Add New Order Item</h4>
```
Comment (HassanAkbar) [inline app/views/orders/_searchform.html.erb]: change id name or remove it if not required

File: app/views/orders/index.html.erb [hunk 10]
```diff
@@ -1,34 +1,69 @@
...
+<table class="table table-bordered">
   <tr>
-    <th>Staff email</th>
+    <th colspan="7" id="table-heading">
+      <div class="row">
+        <div class="col-lg-2">
+          <div class="text-left">
+            <%= select_tag(
+                'search_field',
+                options_for_select([['Search by Waiter', 'waiter'],['Search by Status','status']]),
```
Comment (HassanAkbar) [inline app/views/orders/index.html.erb]: move this to helper.
space before 'status'

File: .gitignore [hunk 11]
```diff
@@ -9,6 +9,7 @@
 
 # Ignore all logfiles and tempfiles.
 !/log/.keep
```
Comment (sehrnadeem1) [inline .gitignore:11]: dont push

File: app/controllers/items_controller.rb [hunk 12]
```diff
@@ -1,42 +1,70 @@
 class ItemsController < ApplicationController
   load_and_authorize_resource
 
-  #POST /items/:id
+  #GET /items
+  def index
+    @items = @items.order(created_at: :desc).paginate(page: params[:page], per_page: LISTING_PAGINATION_SIZE)
+    @items = @items.where(active: true)
```
Comment (sehrnadeem1) [inline app/controllers/items_controller.rb:7]: change

File: app/controllers/orders_controller.rb [hunk 13]
```diff
@@ -1,68 +1,96 @@
...
-      @orders = @user.orders
-      respond_to do |format|
-        format.html
-      end
-    else
-      respond_to do |format|
-        format.html
+    if params[:search_field].present? && params[:search_value].present?
+      if params[:search_field] == 'waiter'
+        @orders = Order.where(user_id: params[:search_value].to_i)
```
Comment (sehrnadeem1) [inline app/controllers/orders_controller.rb:9]: @orders.where

**Pull request merged**